osquery/tables/system/windows/programs.cpp

```diff
@@ -27,47 +27,49 @@ QueryData genPrograms(QueryContext& context) {
   QueryData results;
   QueryData regResults;
   queryKey(
-      "HKEY_LOCAL_MACHINE",
-      "SOFTWARE\\Wow6432Node\\Microsoft\\Windows\\CurrentVersion\\Uninstall",
+      "HKEY_LOCAL_MACHINE\\SOFTWARE\\Wow6432Node\\Microsoft\\"
+      "Windows\\CurrentVersion\\Uninstall",
       regResults);
   for (const auto& rKey : regResults) {
-    QueryData appResults;
-    std::string subkey = rKey.at("subkey");
-    // make sure it's a sane uninstall key
-    boost::smatch matches;
-    boost::regex expression(
+    if (rKey.at("type") == "subkey") {
```

Comment:
Maybe instead of checking `rKey.at("type") == ` we could check `rKey.at("type") != ` and `continue;` if true? That way we could avoid the increased scope depth below? Same note for the above instances.